test/integration/host_test.rb

```diff
@@ -533,6 +475,29 @@ def test_show_action_buttons
       host.reload
       assert_equal env1.name, host.environment.name
     end
+
+    test 'enables override of hostgroup defaults' do
+      original_environment = FactoryGirl.create(:environment)
+      original_hostgroup = FactoryGirl.create(:hostgroup,
+                                              :environment => original_environment)
+
+      # Clear host environment, then inherit it from hostgroup
+      @host.environment = nil
+      @host.hostgroup = original_hostgroup
+      @host.send(:assign_hostgroup_attributes, ['environment_id'])
+      @host.save
+
+      overridden_hostgroup = FactoryGirl.
+        create(:hostgroup, :environment => FactoryGirl.create(:environment))
+
+      visit edit_host_path(@host)
```

Comment:
I'm unsure then from the test title what it should be testing.  It seems to test only the host group change behaviour, not really the override (which wouldn't require a hg change).

Perhaps you can reply to my earlier comment about this then, to say whether the behaviour change in 77b36c9 in the new host case was intended or not.  The change seems OK to me, but the test expectation would need changing if it was to stay against the new host.  Perhaps we should have both, with different expectations - or the behaviour change is a bug, and we need both with the same expectation.